transform-plugins/src/main/java/io/cdap/plugin/Decryptor.java

```diff
@@ -79,6 +83,29 @@ public void initialize(TransformContext context) throws Exception {
     fieldEncryptor.initialize();
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) throws Exception {
+    super.prepareRun(context);
+    decryptFields = conf.getDecryptFields();
+    if (context.getInputSchema() == null || context.getInputSchema().getFields() == null) {
+      return;
+    }
+
+    // Use all decryptFields from conf that also exist in input schema.
+    //    Transform only modifies those fields, and has a null check, but we don't know the field values here.
+    List<String> fields = TransformLineageRecorderUtils.getFields(context.getInputSchema()).stream()
+      .filter(decryptFields::contains)
+      .collect(Collectors.toList());
+
+    List<String> idFields = TransformLineageRecorderUtils.getFields(context.getInputSchema());
+    idFields.removeAll(fields);
+
+    context.record(TransformLineageRecorderUtils.eachInToSomeOut(fields, fields, idFields,
+      "decrypt", "Decrypted the requested fields.",
+      "", "",
```

Comment:
just separate the util function and call the one you need, same applies to other places